### DISCUSSION TITLE
ca.pem not found, self signed certificate, proxy server

### DISCUSSION THREAD

**Discussion by KotlinIsland** [renovatebot/renovate#9545] [Request Help]
**What Renovate type, platform and version are you using?**
self hosted, Gitlab, 24.114.2

**Describe the bug**
build.gradle.kts file is not being detected.

I'm under a corporate proxy with self signed certificates. I think docker images are having issues with either the proxy settings or the certs.

https://github.com/renovatebot/renovate/issues/2835#issuecomment-725834203 seems to relate to this and mentions self hosting the images and setting the `dockerImagePrefix`.

Is that whats happening here? and if so which image(s) need to be hosted?

Also getting a warning about dangling containers.  

**Relevant debug logs**
```
 WARN: Gradle extraction failed (repository=...)
       "errMessage": "Command failed: docker pull docker.io/renovate/gradle\nunable to resolve docker endpoint: open /certs/client/ca.pem: no such file or directory\n"
```
```
WARN: Error removing dangling containers
       "err": {
         "killed": false,
         "code": 1,
         "signal": null,
         "cmd": "docker ps --filter label=renovate_child -aq",
         "stdout": "",
         "stderr": "unable to resolve docker endpoint: open /certs/client/ca.pem: no such file or directory\n",
         "message": "Command failed: docker ps --filter label=renovate_child -aq\nunable to resolve docker endpoint: open /certs/client/ca.pem: no such file or directory\n",
         "stack": "Error: Command failed: docker ps --filter label=renovate_child -aq\nunable to resolve docker endpoint: open /certs/client/ca.pem: no such file or directory\n\n    at ChildProcess.exithandler (child_process.js:308:12)\n    at ChildProcess.emit (events.js:315:20)\n    at ChildProcess.EventEmitter.emit (domain.js:467:12)\n    at maybeClose (internal/child_process.js:1048:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)"
       }
```

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [x] The nature of this bug means it's impossible to reproduce publicly

**Comment by rarkins** [comment]
Please provide more details on exactly how you are running Renovate.

**Reply by KotlinIsland** [reply]
@rarkins Thanks!

I followed https://gitlab.com/renovate-bot/renovate-runner/README.md

My `.gitlab.yml` looks like:
```yaml
variables:
  HTTP_PROXY: ...
  HTTPS_PROXY: ...
  NODE_TLS_REJECT_UNAUTHORIZED: "0"
  GIT_SSL_NO_VERIFY: "1"

include:
  - project: 'this-group/renovate-runner'
    file: '/templates/renovate-dind.gitlab-ci.yml'

renovate:
  only:
    - schedules
    - pushes
    - web
```
And just copied the templates, I removed the line `RENOVATE_EXTENDS: 'github>whitesource/merge-confidence:beta'` because it made it fail.

ci variable:
`--autodiscover=true --autodiscover-filter=this-group/**`

**Reply by viceice** [reply]
You can't currently use docker sidecar images with custom certs. You would need to add the certs to all required sidecar images in advace and use a custom dockerImagePrefix to point to your images.

A easier way is to use the full image and configure custom ca certs via env and volume, checkout this [docs](https://github.com/containerbase/buildpack/blob/main/docs/custom-root-ca.md#runtime-install)

**Reply by viceice** [reply]
You need a GITHUB readonly token to use `github>whitesource/merge-confidence:beta` preset
